index.d.ts

```diff
@@ -11,13 +11,16 @@ export type BeforeRequestHook = (
 	options: NormalizedOptions,
 ) => Request | Response | void | Promise<Request | Response | void>;
 
-export type BeforeRetryHook = (options: {
+export type BeforeRetryOptions = {
```

Comment:
I don't think "options" is the correct word here.

Reply:
What one then? Maybe Params?

Reply:
It's better to present solutions than asking questions. Try to think of some good names. I don't think `Params` is a good name. We try not to use abbreviations here.

Reply:
Well, I can't think of a better name than "parameters". The properties of this object don't have anything in common so that there could be a sensible name for them as a whole, it's there just to reduce amount of arguments for `beforeRetry`. Params seems like a very common abbreviation, but it could be `BeforeRetryParameters`. If not this word, I don't know what one tbh

Reply:
I would name it `BeforeRetryState`.

If not `State`, then some other ideas...
 - `Context`
 - `Record`
 - `Variables` (or `Vars`)
 - `Properties` (or `Props`)
 - `Arguments` (or `Args`)

Reply:
> Well, I can't think of a better name tha "parameters".

Naming is indeed hard, but I don't think you tried your best here. All I asked was to come up with some alternatives.

Reply:
I prefer `Context`, but `State` would work too.

Reply:
> Naming is indeed hard, but I don't think you tried your best here.

Well, sorry for not doing my best for someone else's library. I'll change it to `State`